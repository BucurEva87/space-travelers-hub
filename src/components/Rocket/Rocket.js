import React from 'react';
import { PropTypes as PT } from 'prop-types';
import { useDispatch } from 'react-redux';
import { bookRocket, unbookRocket } from '../../redux/rockets/rockets';
import style from './Rocket.module.scss';

const Rocket = ({
  id, name, description, flickrImages, reserved,
}) => {
  const dispatch = useDispatch();

  return (
    <article data-name={name} data-id={id} className={style.rocket}>
      <img src={flickrImages} alt={description} />
      <div className={style['rocket-info']}>
        <h2 className={style.title}>{name}</h2>
        <p className={style.description}>
          { reserved
          && <span className={style['reserved-tag']}>Reserved</span> }
          {' '}
          {description}
        </p>
        { reserved ? (
          <button
            className={style['btn-unreserve']}
            type="button"
            onClick={(e) => dispatch(unbookRocket(+e.target.closest('article').dataset.id))}
          >
            Cancel Reservation
          </button>
        ) : (
          <button
            className={style['btn-reserve']}
            type="button"
            onClick={(e) => dispatch(bookRocket(+e.target.closest('article').dataset.id))}
          >
            Reserve Rocket
          </button>
        )}
      </div>
    </article>
  );
};

Rocket.propTypes = {
  id: PT.number.isRequired,
  name: PT.string.isRequired,
  description: PT.string.isRequired,
  flickrImages: PT.string.isRequired,
  reserved: PT.bool.isRequired,
};

export default Rocket;
