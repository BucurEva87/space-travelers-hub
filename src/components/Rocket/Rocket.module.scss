@import '../../styles/variables';

.rocket {
  display: flex;
  gap: 1.6rem;
}

.rocket img {
  width: 20%;
}

.info {
  display: flex;
  flex-direction: column;
  font-family: $montserrat;
}

.title {
  font-size: 1.4rem;
  margin-bottom: 1rem;
}

.description {
  margin-bottom: 1rem;
  font-size: 1rem;
}

.btn-reserve {
  background-color: $primaryBtnBlue;
  border: none;
  padding: 0.6rem 1rem;
  color: $white;
  border-radius: 0.4rem;
  cursor: pointer;
}

.btn-unreserve {
  background-color: $white;
  border: 0.1rem solid $gray;
  padding: 0.6rem 1rem;
  color: $gray;
  border-radius: 0.4rem;
  cursor: pointer;
}
