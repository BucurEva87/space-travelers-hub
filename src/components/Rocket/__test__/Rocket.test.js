import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import store from '../../../redux/configureStore';
import Rocket from '../Rocket';

let getAllByRole;

const rockets = [
  {
    id: 1,
    name: 'Falcon 1',
    description: 'The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.',
    flickrImages: 'https://imgur.com/DaCfMsj.jpg',
    wikipedia: 'https://en.wikipedia.org/wiki/Falcon_1',
    reserved: false,
  },
  {
    id: 2,
    name: 'Falcon 9',
    description: 'Falcon 9 is a two-stage rocket designed and manufactured by SpaceX for the reliable and safe transport of satellites and the Dragon spacecraft into orbit.',
    flickrImages: 'https://farm1.staticflickr.com/929/28787338307_3453a11a77_b.jpg',
    wikipedia: 'https://en.wikipedia.org/wiki/Falcon_9',
    reserved: true,
  },
];

beforeEach(() => {
  const component = render(
    <Provider store={store}>
      <Rocket
        key={rockets[0].id}
        id={rockets[0].id}
        name={rockets[0].name}
        description={rockets[0].description}
        flickrImages={rockets[0].flickrImages}
        wikipedia={rockets[0].wikipedia}
        reserved={rockets[0].reserved}
      />
      <Rocket
        key={rockets[1].id}
        id={rockets[1].id}
        name={rockets[1].name}
        description={rockets[1].description}
        flickrImages={rockets[1].flickrImages}
        wikipedia={rockets[1].wikipedia}
        reserved={rockets[1].reserved}
      />
    </Provider>,
  );

  getAllByRole = component.getAllByRole;
});

describe('Rocket component', () => {
  test('is rendered two times on the screen', () => {
    expect(getAllByRole('article')).toHaveLength(2);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <Rocket
          key={rockets[0].id}
          id={rockets[0].id}
          name={rockets[0].name}
          description={rockets[0].description}
          flickrImages={rockets[0].flickrImages}
          wikipedia={rockets[0].wikipedia}
          reserved={rockets[0].reserved}
        />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
