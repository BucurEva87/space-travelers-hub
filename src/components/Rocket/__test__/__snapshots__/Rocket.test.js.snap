// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rocket component matches snapshot 1`] = `
<article
  className="rocket"
  data-id={1}
  data-name="Falcon 1"
>
  <img
    alt="The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth."
    src="https://imgur.com/DaCfMsj.jpg"
  />
  <div
    className="rocket-info"
  >
    <h2
      className="title"
    >
      Falcon 1
    </h2>
    <p
      className="description"
    >
       
      The Falcon 1 was an expendable launch system privately developed and manufactured by SpaceX during 2006-2009. On 28 September 2008, Falcon 1 became the first privately-developed liquid-fuel launch vehicle to go into orbit around the Earth.
    </p>
    <button
      className="btn-reserve"
      onClick={[Function]}
      type="button"
    >
      Reserve Rocket
    </button>
  </div>
</article>
`;
